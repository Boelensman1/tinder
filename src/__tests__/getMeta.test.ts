import * as nock from 'nock';
import * as fs from 'fs';

import TinderClient from '..'; // tslint:disable-line import-name
import { token, id } from './util/getAuth';

nock.load('./src/__tests__/mocks/auth.json');
nock.load('./src/__tests__/mocks/getMeta.json');

const tinderClient = new TinderClient();

beforeAll(() => (
  tinderClient.auth(token, id)
));

it('gets meta information', async () => {
  const meta = await tinderClient.getMeta();
  expect(meta).toMatchSnapshot();
});
