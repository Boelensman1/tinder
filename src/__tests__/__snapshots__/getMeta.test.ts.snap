// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gets meta information 1`] = `
UserMetaData {
  "clientResources": Object {
    "plus_screen": Array [
      "unlimited_swipes",
      "who_you_see",
      "who_sees_you",
      "your_profile",
      "super_like",
      "passport",
      "undo",
      "hide_ads",
    ],
    "rate_card": Object {
      "carousel": Array [
        Object {
          "slug": "unlimited_likes",
        },
        Object {
          "slug": "boost",
        },
        Object {
          "slug": "who_sees_you",
        },
        Object {
          "slug": "your_profile",
        },
        Object {
          "slug": "super_like",
        },
        Object {
          "slug": "passport",
        },
        Object {
          "slug": "undo",
        },
        Object {
          "slug": "no_ads",
        },
      ],
    },
  },
  "globals": Object {
    "ad_swipe_interval": 20,
    "app_boy": true,
    "boost": true,
    "boost_decay": 180,
    "boost_down": 8,
    "boost_duration": 1800000,
    "boost_intro_multiplier": 10,
    "boost_up": 7,
    "can_add_photos_from_facebook": true,
    "can_edit_jobs": false,
    "can_edit_schools": false,
    "can_show_common_connections": true,
    "card_replay": false,
    "discount": true,
    "fast_match": true,
    "fast_match_boost_new_count_fetch_interval": 2000,
    "fast_match_new_count_fetch_interval": 10000,
    "fast_match_new_count_threshold": 99,
    "fast_match_notif_default": 1,
    "fast_match_notif_options": Array [
      0,
      1,
      3,
      10,
    ],
    "fast_match_polling_mode": 1,
    "fast_match_preview_minimum_time": 180000,
    "fetch_connections": false,
    "fireboarding": true,
    "invite_type": "client",
    "kontagent": false,
    "kontagent_enabled": false,
    "matchmaker_default_message": "I want you to meet someone. I introduced you on Tinder www.gotinder.com/app",
    "mqtt": false,
    "photo_preview_enabled": false,
    "plus": true,
    "rate_app": true,
    "recs_blend": false,
    "recs_interval": 20000,
    "recs_size": 40,
    "share_default_text": "<style>body{color:#fff;text-align:center;font-family:HelveticaNeue;text-shadow:0 1px 1px rgba(0,0,0,0.63);}h1{font-size:24px;line-height:24px;margin:0;}p{font-size:16px;margin:8px;}</style><h1>Get a Boost</h1><p><strong>Invite friends</strong> to show up <br/><strong>even higher</strong> in recommendations.</p>",
    "sparks": false,
    "sparks_enabled": false,
    "squads_enabled": false,
    "squads_expiration_notice": 21600000,
    "squads_extension_length": 86400000,
    "super_like": true,
    "super_like_alc_mode": 2,
    "tinder_sparks": true,
    "updates_interval": 2000,
    "web_payments_enabled": true,
  },
  "groups": Array [
    Object {
      "group_id": "hidden",
      "key": "plus_subscription_1499_12m_1v",
      "sub_type": "plus",
      "type": "purchase",
      "version": 0,
    },
    Object {
      "group_id": "hidden",
      "is_primary": true,
      "key": "plus_subscription_1499_6m_1v",
      "sub_type": "plus",
      "type": "purchase",
      "version": 0,
    },
    Object {
      "group_id": "hidden",
      "is_base_group": true,
      "key": "plus_subscription_1499_1m_1v",
      "sub_type": "plus",
      "type": "purchase",
      "version": 0,
    },
  ],
  "notifications": Array [],
  "products": Object {
    "boost": Object {
      "regular": Object {
        "skus": Array [
          Object {
            "amount": 10,
            "product_id": "boost_consumable_2499_10",
            "product_type": "boost",
            "purchase_type": "consumable",
          },
          Object {
            "amount": 5,
            "is_primary": true,
            "product_id": "boost_consumable_1499_5",
            "product_type": "boost",
            "purchase_type": "consumable",
          },
          Object {
            "amount": 1,
            "is_base_group": true,
            "product_id": "boost_consumable_399_1",
            "product_type": "boost",
            "purchase_type": "consumable",
          },
        ],
      },
    },
    "gold": Object {
      "regular": Object {
        "skus": Array [
          Object {
            "product_id": "gold_subscription_1499_12m_1v",
            "product_type": "gold",
            "purchase_type": "subscription",
            "terms": 12,
          },
          Object {
            "is_primary": true,
            "product_id": "gold_subscription_1499_6m_1v",
            "product_type": "gold",
            "purchase_type": "subscription",
            "terms": 6,
          },
          Object {
            "is_base_group": true,
            "product_id": "gold_subscription_1499_1m_1v",
            "product_type": "gold",
            "purchase_type": "subscription",
            "terms": 1,
          },
        ],
      },
    },
    "plus": Object {
      "regular": Object {
        "skus": Array [
          Object {
            "product_id": "plus_subscription_1499_12m_1v",
            "product_type": "plus",
            "purchase_type": "subscription",
            "terms": 12,
          },
          Object {
            "is_primary": true,
            "product_id": "plus_subscription_1499_6m_1v",
            "product_type": "plus",
            "purchase_type": "subscription",
            "terms": 6,
          },
          Object {
            "is_base_group": true,
            "product_id": "plus_subscription_1499_1m_1v",
            "product_type": "plus",
            "purchase_type": "subscription",
            "terms": 1,
          },
        ],
      },
    },
    "superlike": Object {
      "regular": Object {
        "skus": Array [
          Object {
            "amount": 5,
            "is_base_group": true,
            "product_id": "superlike_consumable_499_5",
            "product_type": "superlike",
            "purchase_type": "consumable",
          },
          Object {
            "amount": 25,
            "is_primary": true,
            "product_id": "superlike_consumable_1999_25",
            "product_type": "superlike",
            "purchase_type": "consumable",
          },
          Object {
            "amount": 60,
            "product_id": "superlike_consumable_3999_60",
            "product_type": "superlike",
            "purchase_type": "consumable",
          },
        ],
      },
    },
  },
  "purchases": Array [],
  "rating": Object {
    "likesRemaining": 100,
    "superLikes": Object {
      "alcRemaining": 0,
      "allotment": 1,
      "newAlcRemaining": 0,
      "remaining": 1,
      "resetsAt": null,
      "superlikeRefreshAmount": 5,
      "superlikeRefreshInterval": 1,
      "superlikeRefreshIntervalUnit": "d",
    },
  },
  "travel": Object {
    "isTraveling": false,
  },
  "tutorials": Array [
    "super_like_action",
    "profile_verified",
    "boost_tutorial",
    "welcome_screen",
    "message_standards",
    "reactions_announcement",
    "reactions_tooltip",
    "super_likeable",
    "places_tutorial_cards",
    "places_edit_place_tooltip",
    "loops_new_media_selector",
    "loops_filter_video",
  ],
  "user": User {
    "apiToken": "89ca128e-81d6-468c-a16b-b5ebfb1b06f7",
    "banned": false,
    "bio": "",
    "birthDate": 2000-01-01T00:00:00.000Z,
    "birthDateInfo": undefined,
    "canCreateSquads": false,
    "connectionCount": 100,
    "createDate": 2018-03-23T07:11:14.313Z,
    "discoverable": true,
    "fullName": "AAA BBB",
    "groups": Array [
      "plus_subscription_1499",
    ],
    "id": "89ca128e81d6468ca16bb5eb",
    "isFemale": false,
    "isTraveling": undefined,
    "jobs": undefined,
    "maxDistance": 50,
    "name": "AAA",
    "photos": Array [
      Photo {
        "extension": "jpg",
        "facebookId": "111111111111111",
        "fileName": "89ca128e-81d6-468c-a16b-b5ebfb1b06f7",
        "id": "89ca128e-81d6-468c-a16b-b5ebfb1b06f7",
        "main": true,
        "processedFiles": Array [
          Object {
            "height": 640,
            "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/640x640_89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
            "width": 640,
          },
          Object {
            "height": 320,
            "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/320x320_89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
            "width": 320,
          },
          Object {
            "height": 172,
            "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/172x172_89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
            "width": 172,
          },
          Object {
            "height": 84,
            "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/84x84_89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
            "width": 84,
          },
        ],
        "tinderClient": TinderClient {
          "http": HttpClient {
            "authToken": "89ca128e-81d6-468c-a16b-b5ebfb1b06f7",
            "axiosClient": [Function],
          },
        },
        "url": "https://images-ssl.gotinder.com/89ca128e81d6468ca16bb5eb/89ca128e-81d6-468c-a16b-b5ebfb1b06f7.jpg",
      },
    ],
    "pingTime": 2018-03-23T08:11:14.332Z,
    "schools": Array [
      Object {
        "displayed": true,
        "id": "111111111111111",
        "name": "School",
        "type": "College",
      },
    ],
    "spotifyThemeTrack": undefined,
    "squadAdsShown": undefined,
    "squads": Array [],
    "tinderClient": TinderClient {
      "http": HttpClient {
        "authToken": "89ca128e-81d6-468c-a16b-b5ebfb1b06f7",
        "axiosClient": [Function],
      },
    },
  },
  "versions": Object {
    "activeText": "0.0.0",
    "ageFilter": "2.1.0",
    "matchmaker": "2.1.0",
    "trending": "10.0.0",
    "trendingActiveText": "10.0.0",
  },
}
`;
